src/index.js

```diff
@@ -46,6 +46,20 @@ const unexpectedEnzyme = {
       }
     );
 
+    childExpect.exportAssertion(
+      '<ReactWrapper> to have state satisfying <object>',
+      (expect, reactWrapper, state) => {
+        return expect(reactWrapper.state(), 'to satisfy', state);
+      }
+    );
+
+    childExpect.exportAssertion(
+      '<ReactWrapper> [not] to have state <array>',
```

Comment:
I wouldn't expect this to assertion an array of keys.

What about `<ReactWrapper> to have state satisfying <assertion>`. Than you can do the following:

```js
expect(component, 'to have state satisfying to have keys', ['foo', 'bar'])
```